### ISSUE TITLE
more accurate noise generation for 76477 chip

### ISSUE THREAD

**Issue by Piruzzolo** [mamedev/mame#4941]
[Here ](http://www.righto.com/search/label/reverse-engineering#fn:feedback) is an interesting discussion about the chip. In the notes he says he verified the formula for the noise generator (see [here](http://www.righto.com/search/label/reverse-engineering#fn:feedback)). 

Maybe this function should be modified?
https://github.com/mamedev/mame/blob/6f6a87dd452948473b11cb5fb36d1b87f03df2b2/src/devices/sound/sn76477.cpp#L929

**Comment by happppp** [comment]
Looks like the internal logic in MAME is pre-shift, in reverse, and also inversed. But the result sound output will be correct.

**Comment by Piruzzolo** [comment]
is it

**Comment by happppp** [comment]
Might want another pair of eyes to look at it, but yes I think it's equivalent. Also see the note at the top of the file on how the noise function was reverse engineered and that it should match measurements from the real chip.

**Comment by Piruzzolo** [comment]
Yep I see, that's interesting. We could use the other formula if you guys find it more elegant (it is straight from the chip). Al least we should add a note/comment with the chip implementation infos
